python/mlc_chat/chat_module.py

```diff
@@ -202,13 +203,24 @@ class GenerationConfig:
         The temperature applied to logits before sampling. The default value is
         ``0.7``. A higher temperature encourages more diverse outputs, while a
         lower temperature produces more deterministic outputs.
+    presence_penalty : Optional[float]
+        Number between -2.0 and 2.0. Positive values penalize new tokens based on
+        whether they appear in the text so far, increasing the model's likelihood
+        to talk about new topics. Negative values can increase the likelihood of
+        repetition.
+    frequency_penalty : Optional[float]
+        Number between -2.0 and 2.0. Positive values penalize new tokens based on their
+        existing frequency in the text so far, decreasing the model's likelihood to
+        repeat the same line verbatim. Negative values can increase the likelihood of
+        repetition.
     repetition_penalty : Optional[float]
```

Comment:
Hi, thank you for the great work, @Kartik14! May I ask what is the difference between `frequency_penalty` and `repetition_penalty`? `repetition_penalty` does not seem to be in openai spec.